src/main/java/ch/njol/skript/expressions/ExprLastDeath.java

```diff
@@ -0,0 +1,137 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.Statistic;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Date;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Last Death")
+@Description({"The time of the last death of a player.", 
+				"A change to this value can't be negative, so it will be reset or remain unchanged."})
+@Examples("send \"Your last death was %last death of player%!\" to player")
+@Since("INSERT VERSION")
+public class ExprLastDeath extends SimplePropertyExpression<Player, Date> {
+
+	private final Statistic LAST_DEATH = Statistic.TIME_SINCE_DEATH;
+
+	static {
+		Skript.registerExpression(ExprLastDeath.class, Date.class, ExpressionType.PROPERTY, 
+				"[the] [time of [the]] last death of %players%",
+				"[the] [time of [the]] %players%'[s] last death"
+		);
+		register(ExprLastDeath.class, Date.class, "[the] last death", "players");
+	}
+
+	@SuppressWarnings({"null", "unchecked"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<Player>) exprs[0]);
+		return true;
+	}
+
+	@Nullable
+	@Override
+	public Date convert(Player player) {
+		Date date = new Date();
+		date.subtract(Timespan.fromTicks_i(player.getStatistic(LAST_DEATH)));
+		return date;
+	}
+
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.DELETE || mode == ChangeMode.REMOVE_ALL)
```

Comment:
i know this way of doing things isn't specific to this PR but when did we stop using switch cases that explicitly look for the valid change modes? this method makes it harder to decipher what is supported not to mention it will cause trouble if we were to add new change modes.

@bensku what do you think about this vs the older switch case style or, for only a couple change modes, simple condition style:
```
if (mode == ChangeMode.SET) {
    return CollectionUtils.Array(Something.class);
}
```
style